src/main/webapp/app/exercises/shared/example-submission/example-submission-import/example-submission-import.component.ts

```diff
@@ -0,0 +1,145 @@
+})
+export class ExampleSubmissionImportComponent implements OnInit {
+    exercise: Exercise;
+    readonly column = TableColumn;
+
+    private search = new Subject<void>();
+    private sort = new Subject<void>();
+
+    submissions: Submission[] = [];
+
+    loading = false;
+    content: SearchResult<Submission>;
+    total = 0;
+    state: PageableSearch = {
+        page: 0,
+        pageSize: 10,
+        searchTerm: '',
+        sortingOrder: SortingOrder.DESCENDING,
+        sortedColumn: TableColumn.ID,
+    };
+
+    constructor(
+        private sortService: SortService,
+        private activeModal: NgbActiveModal,
+        protected jhiAlertService: JhiAlertService,
+        private exampleSubmissionService: ExampleSubmissionService,
+        private pagingService: ExampleSubmissionImportPagingService,
+    ) {}
+
+    ngOnInit(): void {
+        this.content = { resultsOnPage: [], numberOfPages: 0 };
+
+        this.performSearch(this.sort, 0);
+        this.performSearch(this.search, 300);
+    }
+
+    private performSearch(searchSubject: Subject<void>, debounce: number) {
+        searchSubject
+            .pipe(
+                debounceTime(debounce),
+                tap(() => (this.loading = true)),
+                switchMap(() => this.pagingService.searchForSubmissions(this.state, this.exercise.id!)),
+            )
+            .subscribe((resp) => {
+                this.content = resp;
+                this.loading = false;
+                this.total = resp.numberOfPages * this.state.pageSize;
+            });
+    }
+
+    /**
+     * Closes the modal in which the import component is opened by dismissing it
+     */
+    clear() {
+        this.activeModal.dismiss('cancel');
+    }
+
+    /**
+     * Closes the modal in which the import component is opened. Gives the selected submission as a result.
+     *
+     * @param submission The submission which was selected by the user for the import.
+     */
+    openImport(submission: Submission) {
+        this.activeModal.close(submission);
+    }
+
+    set page(page: number) {
+        this.setSearchParam({ page });
+    }
+
+    get page(): number {
+        return this.state.page;
+    }
+
+    sortRows() {
+        this.sortService.sortByProperty(this.content.resultsOnPage, this.sortedColumn, this.listSorting);
+    }
+
+    private setSearchParam(patch: Partial<PageableSearch>) {
+        Object.assign(this.state, patch);
+        this.sort.next();
+    }
+
+    set searchTerm(searchTerm: string) {
+        this.state.searchTerm = searchTerm;
+        this.search.next();
+    }
+
+    get searchTerm(): string {
+        return this.state.searchTerm;
+    }
+
+    set sortedColumn(sortedColumn: string) {
+        this.setSearchParam({ sortedColumn });
+    }
+
+    get sortedColumn(): string {
+        return this.state.sortedColumn;
+    }
+
+    /**
+     * Set the list sorting direction
+     *
+     * @param ascending {boolean} Ascending order set
```

Comment:
maybe the comment "flag indicating if the sorting order is set to ascending or not" is easier to understand?